File: v1/src/main/java/com/api/v1/mappers/RentResponseMapper.java
package com.api.v1.mappers;

import com.api.v1.domain.Car;
import com.api.v1.domain.Customer;
import com.api.v1.domain.Payment;
import com.api.v1.domain.Rent;
import com.api.v1.dtos.RentResponseDto;

public record RentResponseMapper(
        Rent rent,
        Customer customer,
        Car car,
        Payment payment
) {

    public RentResponseDto map() {
        return new RentResponseDto(
                rent.getOrderNumber(),
                new CustomerResponseMapper(customer).mapToDto(),
                new CarResponseMapper(car).mapToDto(),
                new PaymentResponseMapper(payment, payment.getCreditCard()).map(),
                rent.getRentedAt()
        );
    }

}

File: v2/src/main/kotlin/com/api/v2/rent/rent/RentResponseMapper.kt
package com.api.v2.rent.rent

import com.api.v2.car.domain.Car
import com.api.v2.car.dtos.CarResponseDto
import com.api.v2.car.mappers.CarResponseMapper
import com.api.v2.customer.domain.Customer
import com.api.v2.customer.mappers.CustomerResponseMapper
import com.api.v2.payment.domain.Payment
import com.api.v2.payment.mappers.PaymentResponseMapper

class RentResponseMapper {

    companion object {
        fun map(rent: Rent, customer: Customer, car: Car, payment: Payment): RentResponseDto {
            return RentResponseDto (
                rent.orderNumber,
                CustomerResponseMapper.mapToDto(customer),
                CarResponseMapper.mapToDto(car),
                PaymentResponseMapper.map(payment, payment.creditCard),
                rent.rentedAt
            )
        }
    }

}